src/main/kotlin/org/rust/ide/refactoring/RsRenameProcessor.kt

```diff
@@ -31,6 +31,19 @@ class RsRenameProcessor : RenamePsiElementProcessor() {
     override fun canProcessElement(element: PsiElement): Boolean = element is RsNamedElement
 
     override fun renameElement(element: PsiElement, newName: String, usages: Array<out UsageInfo>, listener: RefactoringElementListener?) {
+        if (element is RsPatBinding) {
+            usages
+                .filter {
+                    val usageElement = it.element
+                    usageElement is RsStructLiteralField && usageElement.colon == null
+                }
+                .forEach {
+                    val newPatField = RsPsiFactory(element.project)
+                        .createPatFieldFull(element.text, newName)
```

Comment:
You have to replace the `element` with a new `RsStructLiteralField`, not a `RsPatFieldFull`.

Reply:
Fixed